### ISSUE TITLE
Security Configuration Assessment: add capability for reading output of commands (processes)

### ISSUE THREAD

**Issue by TJOSERAFAEL** [wazuh/wazuh#2681]
### Feature request
|4.0|Sources and Packages|All platforms

The Security Configuration Assessment (SCA) module actually supports reading files, registry values and scanning folders. This is not enough as we can see in this issue https://github.com/wazuh/wazuh/issues/2667.

The capability of reading the output of a command is a must and great enhancement for those use causes where the user needs to check the output.

**Comment by chemamartinez** [comment]
It remains to restrict the execution of remote commands on the agent side with a new internal option. It could prevent to run commands of an incoming policy from the manager without the agent's permission explicitly.

**Comment by Skeptor** [comment]
I've been testing the PR and I got those results:
· Executing a command in Linux agent: OK
· Executing a command with arguments in Linux agent: OK
· Executing two commands with a pipe `|`: Failed. It is caused because the fork created by the system to execute the second command doesn't belong to Wazuh and it can't be readed.
· Executing two commands separated by a semicolon `;`: Failed
· Executing a command in Windows agent: OK
· Executing a command with arguments in Windows agent: Failed

I've been testing this checks to check when could they fail or give incorrect information. For example, if a command is specified without an expected result the check is marked as passed. For example:
`ifconfig` will be marked always as passed while the expected syntax is `command -> regex` ( `ifconfig -> !r:^\sIPv4\s$;` )
This isn't a bug because the command was executed and the check doesn't look for anything but I think that it's a weird behaviour and we should log a message when a check has this form

**Comment by Skeptor** [comment]
After some test on the policy files I've reached some conclusions:
When a command fails, either because it isn't valid or because a system error, the check isn't marked as passed or failed. In this case the check is just ignored and the stored value doesn't change. It is in the roadmap to create a third status for a check: `error`
Also I ran `wazuh-modulesd` with valgrind to search for memory leaks and I got 0 leaks as result. I tried with valid commands and with invalid commands.

In security terms, I've checked that we can use as policy file any file no matter their permissions. This is dangerous because the commands are executed with privileges and executing commands from a file that can be modified by any user in the system means make the system vulnerable to malicious threads.
I think we should disable the execution of commands in any file that can be modified for any other user than root. Maye we can give to the user the chance to monitor any file as an option, but it has to be disable by default.
